beacon_node/rest_api/src/beacon.rs

```diff
@@ -215,13 +214,14 @@ pub fn get_active_validators<T: BeaconChainTypes>(
     };
 
     let mut state = get_state_from_root_opt(&ctx.beacon_chain, state_root_opt)?;
-    state.update_pubkey_cache()?;
 
-    state
-        .validators
+    let validators = state.validators.clone();
```

Comment:
The mutable API requiring us to clone the validators could be a bit costly, but I think it could be worked around in an efficient way, and all this API code is going in the bin anyway

Reply:
> @realbigsean if you use "Resolves #1080" in your issue description then Github will automatically close the issue when the PR is merged.

Nice, thanks for the tip!

Reply:
Nah, I think the best we can do in a situation like this is clone, or create some intermediate list of pubkeys. That said, the way this API endpoint works at the moment is quite weird -- it loads all the validator keys from `state.validators`, and then uses the keys to lookup the validator indices... which it already knew :thinking: I suspect @paulhauner will do something better in the new API (like just loading the pubkeys in order).

Reply:
Yep, we can ignore this since it's fully re-written in the new API.